src/Orleans.Core.Abstractions/IDs/StreamId.cs

```diff
@@ -0,0 +1,176 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+using System.Text;
+using Orleans.Concurrency;
+using Orleans.Streams;
+
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// Identify a Stream within a provider
+    /// </summary>
+    [Immutable]
+    [Serializable]
+    [StructLayout(LayoutKind.Auto)]
+    public readonly struct StreamId : IEquatable<StreamId>, IComparable<StreamId>, ISerializable
```

Comment:
I Would implement a TypeConverter from and to string. I think it could make the live easier as it is supported by some serializers.